cg/store/api/status.py

```diff
@@ -137,387 +136,406 @@ def cases(
+            "case_external_bool": case_data.case_external_bool,
+            "samples_to_receive": case_data.samples_to_receive,
+            "samples_to_prepare": case_data.samples_to_prepare,
+            "samples_to_sequence": case_data.samples_to_sequence,
+            "samples_to_deliver": case_data.samples_to_deliver,
+            "samples_to_invoice": case_data.samples_to_invoice,
+            "samples_received": case_data.samples_received,
+            "samples_prepared": case_data.samples_prepared,
+            "samples_sequenced": case_data.samples_sequenced,
+            "samples_received_at": case_data.samples_received_at,
+            "samples_prepared_at": case_data.samples_prepared_at,
+            "samples_sequenced_at": case_data.samples_sequenced_at,
+            "samples_delivered_at": case_data.samples_delivered_at,
+            "samples_invoiced_at": case_data.samples_invoiced_at,
+            "case_action": case_data.case_action,
+            "analysis_completed_at": case_data.analysis_completed_at,
+            "analysis_uploaded_at": case_data.analysis_uploaded_at,
+            "samples_delivered": case_data.samples_delivered,
+            "analysis_delivery_reported_at": case_data.analysis_delivery_reported_at,
+            "samples_invoiced": case_data.samples_invoiced,
+            "analysis_pipeline": case_data.analysis_pipeline,
+            "samples_received_bool": case_data.samples_received_bool,
+            "samples_prepared_bool": case_data.samples_prepared_bool,
+            "samples_sequenced_bool": case_data.samples_sequenced_bool,
+            "analysis_completed_bool": case_data.analysis_completed_bool,
+            "analysis_uploaded_bool": case_data.analysis_uploaded_bool,
+            "samples_delivered_bool": case_data.samples_delivered_bool,
+            "analysis_delivery_reported_bool": case_data.analysis_delivery_reported_bool,
+            "samples_invoiced_bool": case_data.samples_invoiced_bool,
+            "flowcells_status": case_data.flowcells_status,
+            "flowcells_on_disk": case_data.flowcells_on_disk,
+            "flowcells_on_disk_bool": case_data.flowcells_on_disk_bool,
+            "tat": case_data.tat,
+            "is_rerun": case_data.is_rerun,
+            "max_tat": case_data.max_tat,
+        }
+        return case
 
-        # other joins
-        families_q = families_q.outerjoin(
-            models.Family.analyses, models.Sample.invoice, models.Sample.flowcells
+    @staticmethod
+    def _should_be_skipped(
+        case_data,
+        exclude_analysed,
+        exclude_delivered,
+        exclude_delivery_reported,
+        exclude_invoiced,
+        exclude_prepared,
+        exclude_received,
+        exclude_sequenced,
+        exclude_uploaded,
+        only_analysed,
+        only_delivered,
+        only_delivery_reported,
+        only_invoiced,
+        only_prepared,
+        only_received,
+        only_sequenced,
+        only_uploaded,
+    ):
```

Comment:
could this perhaps be handled as a sql query? Filtering query if skip term is present.

Reply:
thanks for reviewing!

Reply:
These are calculated data, to determine if the case should be skipped because of received, we loop the samples and check the attributes on them, how do you propose creating a query like that?